File: server/src/main/java/com/auroratms/tournamentevententry/OriginalEntryInfo.java
package com.auroratms.tournamentevententry;

import com.auroratms.tournamententry.MembershipType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;
import java.util.Map;

/**
 * Tournament entry and event information prior to any changes to restore after
 * user chooses to discard current cart session changes
 */
@NoArgsConstructor
@Getter
@Setter
public class OriginalEntryInfo {
    private Long entryId;
    private String cartSessionId;
    private boolean withdrawing;
    private MembershipType membershipType;
    private int usattDonation;
    private Map<Long, String> doublesEventToPartnerMap;
}

File: client/src/app/tournament/tournament-entry/model/original-entry-info.ts
import {MembershipType} from './tournament-entry.model';

/**
 * Class representing information needed to restore entry to its previous state
 */
export class OriginalEntryInfo {
  // tournament entry id
  entryId: number;

  // cart session with which the modified entries are associated
  cartSessionId: string;

  // if true we are withdrawing entirely from the tournament
  withdrawing: boolean;

  // original membership type
  membershipType: MembershipType;

  // original USATT donation
  usattDonation: number;

  doublesEventToPartnerMap: any;
}
